api/src/opentrons/protocol_runner/legacy_command_mapper.py

```diff
@@ -192,78 +229,223 @@ def _build_initial_command(
+                    startedAt=now,
+                    params=pe_commands.AspirateParams.construct(
+                        pipetteId=pipette_id,
+                        labwareId=labware_id,
+                        wellName=well_name,
+                        volume=volume,
+                        flowRate=flow_rate,
+                    ),
+                )
+            else:
+                flow_rate = command["payload"]["rate"] * pipette.flow_rate.dispense
+                return pe_commands.Dispense.construct(
+                    id=command_id,
+                    key=command_id,
+                    status=pe_commands.CommandStatus.RUNNING,
+                    createdAt=now,
+                    startedAt=now,
+                    params=pe_commands.DispenseParams.construct(
+                        pipetteId=pipette_id,
+                        labwareId=labware_id,
+                        wellName=well_name,
+                        volume=volume,
+                        flowRate=flow_rate,
+                    ),
+                )
+        else:
+            # TODO:(jr, 15.08.2022): aspirate and dispense commands with no specified labware
+            # get filtered into custom. Refactor this in followup legacy command mapping
+            return pe_commands.Custom.construct(
                 id=command_id,
                 key=command_id,
                 status=pe_commands.CommandStatus.RUNNING,
                 createdAt=now,
                 startedAt=now,
-                params=pe_commands.PickUpTipParams.construct(
-                    pipetteId=pipette_id,
-                    labwareId=labware_id,
-                    wellName=well_name,
+                params=LegacyCommandParams.construct(
+                    legacyCommandType=command["name"],
+                    legacyCommandText=command["payload"]["text"],
                 ),
             )
-        elif (
-            command["name"] == legacy_command_types.DROP_TIP
-            and "instrument" in command["payload"]
-            and "location" in command["payload"]
-        ):
-            pipette: LegacyPipetteContext = command["payload"]["instrument"]  # type: ignore
-            location = command["payload"].get("location")
-            if isinstance(location, Location):
+
+    def _build_blow_out_command(
+        self,
+        command: legacy_command_types.BlowOutMessage,
+        command_id: str,
+        now: datetime,
+    ) -> pe_commands.Command:
+        pipette: LegacyPipetteContext = command["payload"]["instrument"]
+        location = command["payload"]["location"]
+        flow_rate = pipette.flow_rate.blow_out
+        #   TODO:(jr, 15.08.2022): blow_out commands with no specified labware get filtered
+        #   into custom. Remove location.labware.is_empty is False when refactor is complete
+        if isinstance(location, Location) and location.labware.is_empty is False:
+            if location.labware.is_well:
```

Comment:
Should this just be one `if`?

```suggestion
        if isinstance(location, Location) and location.labware.is_well:
```